src/Extractor/LinkCollectionExtractorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\Hal\Extractor;
+
+use Zend\Stdlib\Extractor\ExtractionInterface;
+
+interface LinkCollectionExtractorInterface extends ExtractionInterface
+{
+    /**
+     * @return LinkExtractorInterface
+     */
+    public function getLinkExtractor();
+
+    /**
+     * @param LinkExtractorInterface $linkExtractor
+     */
+    public function setLinkExtractor(LinkExtractorInterface $linkExtractor);
```

Comment:
I don't now. I added the setter to improve flexibility but it is maybe not essential.